### DISCUSSION TITLE
Why `$i18n` scope is different between composition API and Legacy API

### DISCUSSION THREAD

**Discussion by LasyIsLazy** [intlify/vue-i18n#993] [Q&A]
In Legacy mode, `$i18n` always refers to the local scope instance,  but in Composition mode, it seems that `$i18n` always refers to the global composer. It's confusing `$i18n` refers to different scope when the mode change.

**Comment by kazupon** [comment]
This is Composition API limitation.

Although the Legacy API in vue-i18n behaves as much as possible like the previous vue-i18n v8.x for APIs prefixed with `$`, such as `$i18n`, in order to provide `$` prefixed methods and computed props in Composition API, you need to use `app.config.globalProperties`.
That is, Vue injects them globally in order to use them in any component.
vue-i18n is necessarily affected by this constraint.

**Reply by parallels999** [reply]
@LasyIsLazy is this **answered**?
